### PR TITLE
feat: Separate admin page [WEB-720]

### PR DESCRIPTION
**Pull request by gt2345**
## Description

Add a new page for user management and group management.

## Test Plan

Log in as admin, click avatar -> "Admin"
"Users" and "Groups" tabs should show up and work as expected.
Log in as non-admin, click avatar, "Admin" tab should not be visible.  

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/components/NavigationSideBar.tsx [hunk 1]
```diff
@@ -196,9 +196,25 @@ const NavigationSideBar: React.FC = () => {
   }, [openWorkspaceCreateModal]);
 
   const pinnedWorkspaces = useWorkspaces({ pinned: true });
+  const { canAdministrateUsers } = usePermissions();
 
   if (!showNavigation) return null;
 
+  const menuItems = [
```
Comment (keita-determined) [inline webui/react/src/components/NavigationSideBar.tsx:203]: could you add a type?
```tsx
const menuItems: MenuProps['items'] = ...
```

File: webui/react/src/pages/Admin/index.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+export type TabType = ValueOf<typeof TabType>;
+
+type Params = {
+  tab?: TabType;
+};
+
+const TAB_KEYS = {
+  [TabType.UserManagement]: 'user-management',
+  [TabType.GroupManagement]: 'group-management',
+};
```
Comment (keita-determined) [inline webui/react/src/pages/Admin/index.tsx:29]: could you add `as const`?

File: webui/react/src/pages/Admin/index.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+const DEFAULT_TAB_KEY = TabType.UserManagement;
+
+const SettingsContent: React.FC = () => {
+  const navigate = useNavigate();
+  const { tab } = useParams<Params>();
+  const [tabKey, setTabKey] = useState<TabType>(tab || DEFAULT_TAB_KEY);
+  const { updateSettings } = useSettings<object>({
```
Comment (keita-determined) [inline webui/react/src/pages/Admin/index.tsx:36]: https://github.com/determined-ai/determined/blob/8957c026b344a64b2f868f4fa8b2af7aa89ad5b9/webui/react/src/hooks/useSettings.ts#L23-L27
i think `SettingsConfig` is better than `object`

File: webui/react/src/pages/Admin/index.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      onChange={handleTabChange}
+    />
+  );
+};
+
+const Admin: React.FC = (p) => (
+  <Page bodyNoPadding id="admin" stickyHeader title="Admin Settings">
+    <SettingsContent {...p} />
+  </Page>
+);
```
Comment (keita-determined) [inline webui/react/src/pages/Admin/index.tsx:92]: do we need this?
how come `SettingsContent` and `Admin` components are split?

File: webui/react/src/routes/routes.ts [hunk 5]
```diff
@@ -241,6 +241,19 @@ const routes: RouteConfig[] = [
...
+  {
+    id: 'admin',
+    needAuth: true,
+    path: '/admin/:tab',
+    title: 'Admin',
+  },
+  {
+    id: 'admin',
+    needAuth: true,
+    path: '/admin/',
```
Comment (keita-determined) [inline webui/react/src/routes/routes.ts:253]: nit: i think `/` can be removed

**Review by keita-determined — COMMENTED** [review]
is it a bug?

https://user-images.githubusercontent.com/109113616/207996552-d09f50ba-4675-4086-8a7d-c5b6a4022e91.mov

File: webui/react/src/components/NavigationSideBar.tsx (hunk 1 above)
Comment (gt2345) [reply webui/react/src/components/NavigationSideBar.tsx:203]: What do you mean?

File: webui/react/src/pages/Admin/index.tsx (hunk 3 above)
Comment (gt2345) [reply webui/react/src/pages/Admin/index.tsx:36]: I don't think useSetting is used anyway on this page, removed this part

**Comment by gt2345** [comment]
> is it a bug?
> 

Do you mean once no longer admin, can still see the menu option? 
I did not reproduce this issue, when I made admin no longer "admin", the menu was item no longer visible.

**Comment by keita-determined** [comment]
> > is it a bug?
> 
> @keita-determined Do you mean once no longer admin, can still see the menu option? I did not reproduce this issue, when I made admin no longer "admin", the menu was item no longer visible.

yes. i can reproduce it on Chrome with private mode. What browser are you using?

**Comment by gt2345** [comment]
> > > is it a bug?
> > 
> > 
> > @keita-determined Do you mean once no longer admin, can still see the menu option? I did not reproduce this issue, when I made admin no longer "admin", the menu was item no longer visible.
> 
> yes. i can reproduce it on Chrome with private mode. What browser are you using?

@keita-determined  Got it, apparently, this only happens when the user edited is on the first page. 
Fix is up, but I'm not sure it's worth it.. Included some new states at that page to re-check auth if the user is editing itself.

**Comment by keita-determined** [comment]
> > > > is it a bug?
> > > 
> > > 
> > > @keita-determined Do you mean once no longer admin, can still see the menu option? I did not reproduce this issue, when I made admin no longer "admin", the menu was item no longer visible.
> > 
> > 
> > yes. i can reproduce it on Chrome with private mode. What browser are you using?
> 
> @keita-determined Got it, apparently, this only happens when the user edited is on the first page. Fix is up, but I'm not sure it's worth it.. Included some new states at that page to re-check auth if the user is editing itself.

yaeh that's why i asked if its a bug. i dont think its a critical bug and its not worth putting some effort for this

**Comment by keita-determined** [comment]
oh you already fixed it... thanks!!

**Review by keita-determined — APPROVED** [review]
LGTM

**Pull request merged**